### DISCUSSION TITLE
Why does rendering happen in a separate app?

### DISCUSSION THREAD

**Discussion by MillionIntegrals** [bevyengine/bevy#6151] [Q&A]
It may be entirely clear from the inside, but for a newcomer to bevy, it is a very interesting and highly nontrivial fact to me why does bevy rendering happen in a separate sub-app from the main App - therefore I'm super interested in hearing more about rationale and tradeoffs between having a single app or a sub app.

- Is it just to get clean tracing info on rendering?
- Does it improve parallelism?
  -  because we can render a frame while main app already processes next frame?
  - because we never want the main system to block on vsync (is that even an issue?)
- Does it hurt parallelism?
  - because we can't overlap parts of processing current frame with rendering what's already "ready"
 
From reading bevy source code this seems to be the most nontrivial design decision to me so far, so I'd love to hear some color on it 😄

**Comment by mockersf** [comment]
you can get a few details on the blog post of the 0.6 release, when the new renderer was set up: https://bevyengine.org/news/bevy-0-6/#pipelined-rendering-extract-prepare-queue-render
